### PR TITLE
Moore's voting Algorithm Docs

### PR DESCRIPTION
**Pull request by SAY-droid427**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes https://github.com/TesseractCoding/NeoAlgo-Docs/issues/198

### CODE REVIEW

**Comment by SAY-droid427** [comment]
@iamrajiv @HarshCasper  Please review the PR

**Review by iamrajiv — APPROVED** [review]

File: docs/miscellaneous/moore-voting-algorithm.mdx [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+title: Finding out the majority element using Moore's voting algorithm
```
Comment (HarshCasper) [inline docs/miscellaneous/moore-voting-algorithm.mdx:2]: It should be just `Moore's voting algorithm`

File: docs/miscellaneous/moore-voting-algorithm.mdx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+#### Output
+
+```
+Majority element: 2
+```
+
+## Complexity analysis
+
+- Time Complexity: `O(n)`
+
```
Comment (HarshCasper) [inline docs/miscellaneous/moore-voting-algorithm.mdx:134]: Remove the extra line

File: docs/miscellaneous/moore-voting-algorithm.mdx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+## Code
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', }
```
Comment (HarshCasper) [inline docs/miscellaneous/moore-voting-algorithm.mdx:53]: Can you add another implementation in Java/Python/Golang ?

File: docs/miscellaneous/moore-voting-algorithm.mdx [hunk 4]
```diff
@@ -0,0 +1,140 @@
+---
+title: Finding out the majority element using Moore's voting algorithm
+slug: /miscellaneous/moore-voting-algorithm
+---
+
+## Introduction
+
+The most common algorithm to find the majority element is to keep a count of the frequency of each element in the array and then decide which has the greatest frequency. This has a time complexity of `O(n^2)`. But a faster algorithm exists, called `Moore's Voting algorithm` which reduces the time complexity to `O(n)`. 
+
```
Comment (HarshCasper) [inline docs/miscellaneous/moore-voting-algorithm.mdx:9]: Add an explanation and diagram

File: docs/miscellaneous/moore-voting-algorithm.mdx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+#### Example 1
+
+```
+Let the array be `1,2,2`.
+Initially the majority index is `0`.
+The element at the majority index is equal to the element in position `0`, hence count increments.
+Next element `2` is not the same as the majority index. Count decrements.
+Next element `2` is not same as majority index again. Count decrements to `0`.
+Majority index is changed to `2` and the element at majority index is returned.
+`2` occurs twice, so it is the majority element.
```
Comment (HarshCasper) [inline docs/miscellaneous/moore-voting-algorithm.mdx:31]: Backticks don't work here

File: docs/miscellaneous/moore-voting-algorithm.mdx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+#### Example 2
+
+```
+Let the array be `2,3`.
+Initially the majority index is `0`.
+The element at the majority index is equal to the element in position `0`, hence count increments.
+Next element is not the same as that at the majority index, hence count decrements to `0`.
+Majority index is changed to `1`.
+`3` occurs only once, hence no majority element exists.
```
Comment (HarshCasper) [inline docs/miscellaneous/moore-voting-algorithm.mdx:42]: Backticks don't work here

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by HarshCasper** [comment]
@SAY-droid427 Can you rebase against master and fix the build so that I can give it a final review?

**Comment by SAY-droid427** [comment]
> @SAY-droid427 Can you rebase against master and fix the build so that I can give it a final review?

@HarshCasper Could you please tell me how to do that?

**Comment by SAY-droid427** [comment]
@HarshCasper The build is fixed but one of the checks is failing. Could you please review now?

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**